jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleResolver.java

```diff
@@ -369,16 +369,34 @@ private static final MethodHandle sendResolveMethodHandle(
 			String name,
 			String typeDescriptor,
 			ClassLoader loader) throws Throwable {
-		try {
+		MethodType type = null;
```

Comment:
Why was this moved out of the try block?  Does the OJDK code not need the IllegalAccessException -> IllegalAccessError mapping?

Reply:
> Why was this moved out of the try block? 

It is used by both the implementations. By moving it out, it is shared by both the implementations.

> Does the OJDK code not need the IllegalAccessException -> IllegalAccessError mapping?

Yes, it does need it. The mapping is handled in `linkMethodHandleConstant`, which is OJDK code. So, we don't need to handle the mapping in OpenJ9 code. `linkMethodHandleConstant` invokes `mapLookupExceptionToError` to handle the mapping.

`mapLookupExceptionToError`: https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/ac04db2e39441f285acebb657aa388043a894770/src/java.base/share/classes/java/lang/invoke/MethodHandleNatives.java#L621-L641